rmw_connextdds_common/src/ndds/rmw_type_support_ndds.cpp

```diff
@@ -638,11 +721,349 @@ Key Management functions:
 * -------------------------------------------------------------------------------------- */
 static
 PRESTypePluginKeyKind
-RMW_Connext_TypePlugin_get_key_kind(void)
+RMW_Connext_TypePlugin_get_key_kind_unkeyed(void)
 {
   return PRES_TYPEPLUGIN_NO_KEY;
 }
 
+static
+PRESTypePluginKeyKind
+RMW_Connext_TypePlugin_get_key_kind_keyed(void)
+{
+  return PRES_TYPEPLUGIN_USER_KEY;
+}
+
+static
+RTIBool
+RMW_Connext_TypePlugin_serialize_key(
+  PRESTypePluginEndpointData endpoint_data,
+  const void *key,
+  struct RTICdrStream *stream,
+  RTIBool serialize_encapsulation,
+  RTIEncapsulationId encapsulation_id,
+  RTIBool serialize_key,
+  void *endpoint_plugin_qos)
+{
+  UNUSED_ARG(endpoint_data);
+  UNUSED_ARG(endpoint_plugin_qos);
+
+  if (!serialize_key) {
+    // Not currently supported. As a result, users cannot enable batching by
+    // setting writer_qos.batch.enabled to TRUE.
+    return RTI_FALSE;
+  }
```

Comment:
user still can configure RTI DDS to enable `writer_qos.batch.enabled` and ends up having this failure? if this happens, user can know why it is failing with clear error message from RTI connextdds? if not, it would be nice to add the error message that tells user about this limitation explicitly?

Reply:
See my other [comment](https://github.com/ros2/rmw_connextdds/pull/178#discussion_r2031894358)